### PR TITLE
Feature/program interface

### PR DESCRIPTION
**Pull request by diptorupd**
Adds wrappers for program and kernel. Needs more unit tests to check building program from SPIR-V and the Cython interface.

### CODE REVIEW

**Comment by PokhodenkoSA** [comment]
@diptorupd, could you please merge master to it? It will fix build on CI for this PR.

File: conda-recipe/build.sh [hunk 1]
```diff
@@ -22,31 +25,34 @@ DPCPP_ROOT=${ONEAPI_ROOT}/compiler/latest/linux/
...
+mkdir -p dpctl/include
+cp -r backends/include/* dpctl/include
 
 # required by dpctl.opencl_core
-export DPPL_OPENCL_INTERFACE_LIBDIR=${PREFIX}
-export DPPL_OPENCL_INTERFACE_INCLDIR=${PREFIX}/include
-export OpenCL_LIBDIR=${DPCPP_ROOT}/lib
+export DPPL_OPENCL_INTERFACE_LIBDIR=dpctl
+export DPPL_OPENCL_INTERFACE_INCLDIR=dpctl/include
+export OpenCL_LIBDIR=${DPCPP_ROOT}/libw
```
Comment (reazulhoque) [inline conda-recipe/build.sh:46]: libw -> lib? typo?

Comment (PokhodenkoSA) [reply conda-recipe/build.sh:46]: I think `lib` is correct. It was based on this https://github.com/IntelPython/dpctl/pull/69/commits/24182369db7a69b4e2ef9c21c6e233dad52c453a.

**Comment by diptorupd** [comment]
Superseded by #80. Closing this PR.

**Pull request closed without merging**